odxtools/tablerow.py

```diff
@@ -157,3 +157,8 @@ def structure(self) -> Optional[BasicStructure]:
     def dop(self) -> Optional[DataObjectProperty]:
         """The data object property object resolved by dop_ref."""
         return self._dop
+
+    def __reduce__(self) -> Tuple[Any, ...]:
+        """This ensures that the object can be correctly reconstructed during unpickling."""
+        state = self.__dict__.copy()
+        return self.__class__, tuple([getattr(self, x.name) for x in fields(self)]), state
```

Comment:
is `state` required here?

Reply:
@andlaus I assume for the properties computed through odxlinks

Reply:
@andlaus yes, it's necessary to capture the non-dataclass attributes, such us \_structure, \_key, ...So without including the state, these attributes would not be restored.

Reply:
okay, but this makes serialization of individual objects (i.e., not the whole database) quite a bit larger than strictly necessary because a lot of tangentially related objects (i.e., all referenced objects) will also be serialized? (to be clear: I'm fine with that.)

Reply:
yes you're right, but I think it's better to include the referenced objects in the serialization rather than recomputing them after deserialization.